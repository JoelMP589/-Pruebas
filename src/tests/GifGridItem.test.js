import { shallow } from "enzyme";
import GifGridItem from "../components/GifGridItem";

describe('pruebas en <GifGridItem/>', () => {
    const title = 'Titulo del GIF';
    const url = 'https://media1.giphy.com/media/3o7btLwJWX9hjOZx6w/200w.gif';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de  tener un parrafo con el titulo del gif ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    test('debe de tener una imagen con el url del gif y un alt con el title', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    test('debe de tener una clase "card"', () => {
        expect(wrapper.find('div').hasClass('card')).toBe(true);
    })
})