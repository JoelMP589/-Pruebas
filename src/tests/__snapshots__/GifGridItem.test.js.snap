// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Titulo del GIF"
    src="https://media1.giphy.com/media/3o7btLwJWX9hjOZx6w/200w.gif"
  />
  <p>
    Titulo del GIF
  </p>
</div>
`;
